packages/ibm-products/src/components/Datagrid/Datagrid/addons/Filtering/FilterProvider.js

```diff
@@ -27,22 +34,100 @@ const EventEmitter = {
   },
 };
 
+const removeFilterItem = (state, index) => state.splice(index, 1);
+
+const updateFilterState = (state, type, value) => {
+  if (type === CHECKBOX) {
+    return;
+  }
+  if (type === DATE) {
+    const filterTagIndex = state.findIndex(
+      (val) =>
+        formatDateRange(val.value[0], val.value[1]) ===
+        formatDateRange(value[0], value[1])
+    );
+    return removeFilterItem(state, filterTagIndex);
+  }
+  const filterTagIndex = state.findIndex((val) => val.value === value);
+  return removeFilterItem(state, filterTagIndex);
+};
+
+export const clearSingleFilter = ({ key, value }, setAllFilters, state) => {
+  const tempState = [...state.filters];
+  tempState.forEach((f, filterIndex) => {
+    if (f.id === key) {
```

Comment:
Like in the future I mentioned probably making a hook to encapsulate removing filters, since we have repeated code here and elsewhere